src/command/FileCommands.cxx

```
@@ -234,9 +236,93 @@ handle_read_comments(Client &client, Request args, Response &r)
 	gcc_unreachable();
 }
 
+AllocatedPath
+find_art(AllocatedPath directory)
+{
+    std::vector<std::string> art_names = {
+        "cover.png",
+        "cover.jpg",
+        "cover.tiff",
+        "cover.bmp"
+    };
+    
+    for(std::string& name: art_names) {
+        const AllocatedPath art = AllocatedPath::Build(directory, name.c_str());
+        
+        FileInfo fi;
+        if (!GetFileInfo(art, fi, false)) {
+            continue;
+        }
+        
+        if (!fi.IsRegular()) {
+            continue;
+        }
+        return art;
+    }
+    return AllocatedPath::Null();
+}
+
 CommandResult
 handle_album_art(Client &client, Request args, Response &r)
 {
-    r.Error(ACK_ERROR_SYSTEM, "Unimplemented");
-    return CommandResult::ERROR;
+    assert(args.size == 1 || args.size == 2);
+    
+    const char *const uri = args.front();
+    unsigned long long offset = 0;
+    if (args.size == 2) {
+        offset = args.ParseUnsigned(1);
+    }
+    
+#ifdef ENABLE_DATABASE
+    const Storage *storage = client.GetStorage();
+#endif
+    const auto located_uri = LocateUri(uri, &client
+#ifdef ENABLE_DATABASE
+                                       , storage
+#endif
+                                       );
+    
+    const AllocatedPath path = storage->MapFS(located_uri.canonical_uri);
+    const AllocatedPath directory = path.GetDirectoryName();
+    
+    AllocatedPath art_name = find_art(directory);
+    if (art_name.IsNull()) {
+        if (offset > 0) {
+            r.Error(ACK_ERROR_NO_EXIST, "No art found");
+            return CommandResult::ERROR;
+        }
+        r.Write("size: -1\n");
```

Sorry, I'm not being very clear. I don't think asking for a file that doesn't exist is an error necessarily, as the client won't know until it asks. Therefore I'm not sure the function should return an error. So if we send `album art xyz.mp3` we get back
`size: -1\nOK\n` signalling the file doesn't exist. I don't consider this an error though.

However if the client then asks for `album art xyz.mp3 31268` I think this is an error, and then return `ACK_ERROR_NO_EXIST`. I can change to `ACK_ERROR_ARG` in this scenario if you think thats a better fit.